platform/default/src/mbgl/storage/file_source_manager.cpp

```diff
@@ -0,0 +1,43 @@
+#include <mbgl/storage/asset_file_source.hpp>
+#include <mbgl/storage/database_file_source.hpp>
+#include <mbgl/storage/file_source_manager.hpp>
+#include <mbgl/storage/local_file_source.hpp>
+#include <mbgl/storage/main_resource_loader.hpp>
+#include <mbgl/storage/online_file_source.hpp>
+#include <mbgl/storage/resource_options.hpp>
+
+namespace mbgl {
+
+class DefaultFileSourceManagerImpl final : public FileSourceManager {
```

Comment:
Only issue I see with this is it is a linking time dependency, meaning that if you don't implement `FileSourceManager::get()` it will not link. Would't be better if the platform could simple register the factories or not explicitly during initialization?

Reply:
I'm fine with keeping like this btw and migrate everything later.